src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -696,10 +748,13 @@ void FancyZones::ToggleEditor() noexcept
 
 void FancyZones::SettingsChanged() noexcept
 {
-    std::shared_lock readLock(m_lock);
     // Update the hotkey
     UnregisterHotKey(m_window, 1);
     RegisterHotKey(m_window, 1, m_settings->GetSettings()->editorHotkey.get_modifiers(), m_settings->GetSettings()->editorHotkey.get_code());
+
+    // Needed if we toggled multiMonitorMode
+    m_workAreaHandler.Clear();

```

Comment:
Is there a way (if not, maybe we should enable it) to check which specific setting changed, not to do this for every setting.

Reply:
I think it's OK to do it every time, in general, since saving settings is not something that needs to be optimized for performance, I prefer to have a solid approach that doesn't require maintenance when adding/removing options.